src/main/java/org/testng/GuiceHelper.java

```diff
@@ -41,24 +43,26 @@ Injector getInjector(IClass iClass, IInjectorFactory injectorFactory) {
     }
     Injector parentInjector = ((ClassImpl) iClass).getParentInjector(injectorFactory);
 
-    List<Module> moduleInstances =
-        Lists.newArrayList(getModules(guice, parentInjector, iClass.getRealClass()));
+    List<Module> classLevelModules = getModules(guice, parentInjector, iClass.getRealClass());
     Module parentModule = getParentModule(context);
     if (parentModule != null) {
-      moduleInstances.add(parentModule);
+      context.addGuiceModule(parentModule);
+      BiPredicate<Module, Module> condition = (m,n) -> m.getClass().equals(n.getClass());
```

Comment:
The predicate could be shared between its different usages.

Reply:
Fixed